tools/common/Application.cs

```diff
@@ -490,19 +490,26 @@ public void RunRegistrar ()
 			ps.AssemblyResolver = resolver;
 			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
 
-			var rootName = Path.GetFileNameWithoutExtension (RootAssembly);
-			if (rootName != Driver.GetProductAssembly (this))
-				throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
+			int index = 0;
+			foreach (var asm in RootAssemblies) {
+				var rootName = Path.GetFileNameWithoutExtension (asm);
+				if (index == 0 && rootName != Driver.GetProductAssembly (this))
+					throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
 
-			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse (rootName), new ReaderParameters ()));
-			Driver.Log (3, "Loaded {0}", resolvedAssemblies [resolvedAssemblies.Count - 1].MainModule.FileName);
+				AssemblyDefinition lastAssembly = ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse (rootName), new ReaderParameters ());
```

Comment:
We should consider how to handle 'invalid' assemblies, like the crypto ones. Do we want to write special logic to *not* pass them to mmp, or do we want to process them before crypto. Or do we want to wrap them in a try/catch here and ignore the fact they failed?

Reply:
I agree. Maybe we should file a card or issue and discuss after break when Rolf is around?